zulipterminal/ui_tools/messages.py

```diff
@@ -564,8 +575,15 @@ def soup2markup(
                             continue
                         css_style = code_element.attrs.get("class", ["w"])
                         markup.append((f"pygments:{css_style[0]}", code_text))
+                        code_snippet.append((f"pygments:{css_style[0]}", code_text))
                     else:
                         markup.append(("pygments:w", code_text))
+                        code_snippet.append(("pygments:w", code_text))
+
+                code_language = element.attrs.get("data-code-language")
+
+                if code_language != "Text only":
+                    metadata["code_snippets"].append((code_language, code_snippet))
```

Comment:
It's possible the language data was not available, or was tagged elsewhere, in a previous version of the Zulip markup. This was hinted at in zulip/zulip#11618.

Reply:
When you put `text` after the 3 backticks (```), it causes the code_language to be `"Text only"`. It seems they intended to avoid including that as part of the code snippet for some reason.

I'm looking into where the language data was previously tagged.